docs/profiler.md

```diff
@@ -1,136 +1,196 @@
+* `--threads` (count): Number of CPU threads that will be used to run the benchmark. Generally, more threads mean faster PoST generation and lesser generation time. However, different CPUs behave differently when all threads are under load. Thus, keep experimenting with different values for `--threads` (with the maximum being the number of threads your CPU has) to find the optimal thread count.
+  * **Default**: 4 threads.
+* `--nonces` (count, multiple of 16): The amount of nonces to use in the proof of work calculation in one pass over the PoS data. Read [this section](https://docs.spacemesh.io/docs/learn/post#generating-the-proof) of the PoST explainer to understand how nonces are used in the PoST generation process. The greater the `--nonces` value, the more nonces are used in each data pass, increasing the chance of finding the PoST sooner and in lesser data passes. Note that whatever the number value used for `--nonces`, it must be a multiple of 16. One thing to be aware of when setting `--nonces` is that the higher the value used, the more stress the CPU will be under as this part of the PoST generation process is CPU-intensive.
+  * **Default**: 64 nonces.
+
+#### Nonce estimation formula
+
+To help us estimate the value of `--nonces` that will help find the PoST in the least number of passes, the following formula, given in two prominent formats for ease of calculation, can be used:
+
+**Google Sheets format**:
 ```
-1-(1-(1-BINOM.DIST(36;10^9;26/10^9;TRUE)))^{put nonces here}
+1-(1-(1-BINOM.DIST(36;10^9;26/10^9;TRUE)))^{put nonces value here}
 ```
-in wolfram alpha format
+**Wolfram Alpha format**:
 ```
-1-(1-(1-CDF[BinomialDistribution[10^9, 26/10^9],36]))^{put nonces here}
+1-(1-(1-CDF[BinomialDistribution[10^9, 26/10^9],36]))^{put nonces value here}
 ```
 
-This formula gives the probability of finding a valid proof in one data pass. The node will perform multiple passes if needed.
+The aforementioned formula gives the probability (with an output of 1 being 100%) of finding a valid PoST in one data pass. The node will perform multiple passes if PoST is not found in one data pass.
+
+Please note that PoST generation speed scales *linearly* with the number of CPU cores used (assuming that they are equally fast) and *inversely* with the number of nonces in groups of 16 (32 nonces should be twice as slow as 16). This effect might not manifest itself until a high number of nonces is used. The reason for this is that in most setups, the hard disk drive (HDD) speed will become the limiting factor (as a spinning disk drive is used in most computers) if a low number of nonces is used.
+
+So, if we increase the CPU cores, the faster a valid PoST will be generated. And if you increase the number of nonces, the slower a valid PoST will be generated due to the increased usage of the HDD which results in the HDD speed being the limiting factor. Knowing this will help you make a better assessment of the profiler tool flags and its output.
```

Comment:
increasing nonces requires more CPU power. So the more nonces - the slower it is - that's true, but HDD doesn't become the limiting factor with increasing the nonces - it's the opposite - the CPU becomes the bottleneck. HDD becomes a bottleneck with low nonces and high threads.

Reply:
Reworded to the following:

So, if we increase the CPU cores, the faster a valid PoST will be generated. if we increase the number of nonces, the slower a valid PoST will be generated due to the increased CPU usage which results in the CPU being the limiting factor. On the other hand, with a low number of nonces, the HDD becomes the limiting factor as the PoS data will likely be read more times to find the valid PoST. Knowing this will help us make a better assessment of the profiler tool's inputs and outputs.

Thoughts?

Reply:
With little nonces, the HDD becomes the bottleneck because the CPU is crunching the data faster than the disk can read it. To best use the resource, the user needs to find a balance.

Reply:
Yeah, better :) I'm just little worried that a lazy user will just read the first part:
>So, if we increase the CPU cores, the faster a valid PoST will be generated. if we increase the number of nonces, the slower a valid PoST will be generated 

and set `threads` to max and `nonces` to 1 :)